crates/loam-cli/examples/soroban/nft/src/test.rs

```diff
@@ -0,0 +1,263 @@
+            address: &owner,
+            invoke: &MockAuthInvoke {
+                contract: &contract_id,
+                fn_name: "transfer",
+                args: (nft_id.clone(), owner.clone(), new_owner.clone()).into_val(env),
+                sub_invokes: &[],
+            },
+        }])
+        .transfer(&nft_id, &owner, &new_owner);
+}
+
+#[test]
+fn test_nft() {
+    let (env, client, contract_id) = &init();
+    let admin = Address::generate(&env);
+
+    // test admin_get and admin_set
+    assert!(client.admin_get().is_none(), "Admin already set");
+    set_admin(env, client, contract_id, &admin);
+    assert_eq!(client.admin_get().unwrap(), admin);
+
+    init_nft_contract(env, client, contract_id, &admin, "nftexample");
+    assert_eq!(client.get_total_count(), 0);
+    assert!(client.get_nft(&1).is_none());
+    assert!(client.get_owner(&1).is_none());
+    assert_eq!(client.get_collection_by_owner(&admin).len(), 0);
+
+    // test mint & getter fns
+    let owner_1 = Address::generate(env);
+    let metadata = Bytes::from_slice(env, "metadata".as_bytes());
+    let nft_id = mint_nft(env, client, contract_id, &admin, &owner_1, &metadata);
+    assert_eq!(nft_id, 1);
+    assert_eq!(client.get_nft(&nft_id), Some(metadata));
+    assert_eq!(client.get_owner(&nft_id), Some(owner_1.clone()));
+    assert_eq!(client.get_total_count(), 1);
+    assert_eq!(client.get_collection_by_owner(&owner_1).len(), 1);
+
+    // test transfer
+    let owner_2 = Address::generate(env);
+    transfer_nft(env, client, contract_id, nft_id, &owner_1, &owner_2);
+    assert_eq!(client.get_owner(&nft_id), Some(owner_2.clone()));
+    assert_eq!(client.get_total_count(), 1);
+    assert_eq!(client.get_collection_by_owner(&owner_1).len(), 0);
+    assert_eq!(client.get_collection_by_owner(&owner_2).len(), 1);
+}
+
+#[test]
+#[should_panic(expected = "No admin! Call 'admin_set' first.")]
+fn test_initializing_without_admin_set() {
+    let (env, client, contract_id) = &init();
+    let admin = Address::generate(env);
+    // test nft_init
+    init_nft_contract(env, client, contract_id, &admin, "nftexample");
+}
+
+#[test]
+#[should_panic(expected = "Unauthorized function call for address")]
+fn test_minting_by_non_admin() {
+    let (env, client, contract_id) = &init();
+    let admin = Address::generate(&env);
+
+    // set admin
+    assert!(client.admin_get().is_none(), "Admin already set");
+    set_admin(env, client, contract_id, &admin);
+    assert!(client.admin_get().is_some(), "Admin not set");
+
+    // nft_init
+    init_nft_contract(env, client, contract_id, &admin, "nftexample");
+
+    assert_eq!(client.get_total_count(), 0);
+
+    // try to mint from non-admin
+    let non_admin = Address::generate(env);
+    let metadata = Bytes::from_slice(env, "metadata".as_bytes());
+    mint_nft(env, client, contract_id, &non_admin, &non_admin, &metadata);
+}
+
+#[test]
+#[should_panic]
+#[ignore = "This should panic, but it's not"]
```

Comment:
I expect that these two tests should panic because we are trying to set/get storage state before the contract is initialized. And I though that it would panic because of this https://github.com/loambuild/loam/blob/bc4cf880b95dc82f9005415c5897abe8b92b61c4/crates/loam-soroban-sdk/src/lib.rs#L29

I could use another set of eyes on this in case I'm missing something.